src/Drupal/DrupalKernel.php

```diff
@@ -203,4 +214,77 @@ protected function addDrushServiceProvider($serviceProviderName, $serviceYmlPath
             $this->serviceYamls['app'][$serviceProviderName] = $serviceYmlPath;
         }
     }
+
+    /**
+     * populates theme data on the filesystem.
+     *
+     * @see Drupal\Core\DrupalKernel::moduleData().
+     */
+    protected function themeData($theme_list)
+    {
+        // First, find profiles.
+        $listing = new ExtensionDiscovery($this->root);
+        $listing->setProfileDirectories([]);
+        $all_profiles = $listing->scan('profile');
+        $profiles = array_intersect_key($all_profiles, $theme_list);
+
+        // If a theme is within a profile directory but specifies another
+        // profile for testing, it needs to be found in the parent profile.
+        $settings = $this->getConfigStorage()->read('simpletest.settings');
+        $parent_profile = !empty($settings['parent_profile']) ? $settings['parent_profile'] : null;
+        if ($parent_profile && !isset($profiles[$parent_profile])) {
+            // In case both profile directories contain the same extension, the
+            // actual profile always has precedence.
+            $profiles = array($parent_profile => $all_profiles[$parent_profile]) + $profiles;
+        }
+
+        $profile_directories = array_map(function ($profile) {
+            return $profile->getPath();
+        }, $profiles);
+        $listing->setProfileDirectories($profile_directories);
+
+        // Now find themes.
+        return $listing->scan('theme');
+    }
+
+    /**
+     * Gets the file name for each enabled theme.
+     *
+     * @return array
+     *   Array where each key is a theme name, and each value is a path to the
+     *   respective *.info.yml file.
+     */
+    protected function getThemeFileNames()
+    {
+        if ($this->themeNames) {
+            return $this->themeNames;
+        }
+        $extensions = $this->getConfigStorage()->read('core.extension');
```

Comment:
Lets use a Drupal API like `\Drupal::service('theme_handler')->listInfo();` or `system_list('theme')`

Reply:
Can't do that unless we load core/includes/module.inc. At this point in the bootstrap process, it isn't yet which means `system_list()` is unavailable and `->listInfo()` uses it.